src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,176 @@
+package duke;
+
+import tasks.ToDo;
+import tasks.Deadline;
+import tasks.Event;
+import tasks.TaskList;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+import java.util.Scanner;
+
+/**
+ * Input parsing and displaying output to the user.
+ */
+public class Parser {
+    private final Scanner sc;
```

Comment:
It might make sense to handle `Scanner` object creation and handling within the `Ui` class. Thoughts?